pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -629,7 +651,7 @@ func (c *Controller) setupIndex() *AmbientIndex {
 			updates := handlePod(oldObj, newObj, false)
 			if len(updates) > 0 {
 				c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
-					Full:           false,
+					Full:           true,
```

Comment:
We already have similar/same in https://github.com/istio/istio/blob/7b912eaef96ff7e55ec6b75b457c2b2610bdf863/pilot/pkg/ambient/controller/workloadcache.go#L101. 

We have two code paths:
* Ztunnel (this code you changed), which is efficient
* Waypoint (which I linked), which needs to be rewritten as it is not scalable

I would prefer not to pollute the ztunnel code with incorrect logic if possible, so we can just swap out the impl for waypoint.

Plus it seems weird for a `Address` update needed for waypoint sinceAddress is only used for ztunnel?

Reply:
posted an update on the debug findings in the issue:
https://github.com/istio/istio/issues/42866#issuecomment-1406572832

Reply:
>Plus it seems weird for a Address update needed for waypoint sinceAddress is only used for ztunnel?

@howardjohn the endpoint that needs to be updated is NOT waypoint - the endpoint that needs to be updated is the httpbin endpoint (which stores the waypoint IP in its metadata) stored in gateway/sidecars.

To my knowledge, the fix requires a full push and my logic in `updateEndpointsOnWaypointChange` which needs access to the `kubeEndpointsController` stored in [pilot/pkg/serviceregistry/kube/controller/controller.go](https://github.com/istio/istio/blob/7b912eaef96ff7e55ec6b75b457c2b2610bdf863/pilot/pkg/serviceregistry/kube/controller/controller.go#L235).  This finds all endpoints associated with the service captured by the waypoint in question.

I'm a little unfamiliar with the EDS logic, workloadCache has `model.XDSUpdater.EDSUpdate` and `.EDSCacheUpdate()`, both of which require a shard name and  []model.IstioEndpoint - which didn't sound like the correct path for updating the httpbin endpoints

Reply:
If we are just changing endpoints why do we need a full push? I am against any full push in this code, since we are polluting one of the only production-quality parts of ambient (ztunnel) with un-viable code (full push on every pod) that we will need to dis-entangle later.

Reply:
@howardjohn latest commits have removed setting `full: true` from the logic.  I was triggering a stack trace by not handling both types of endpoint controllers ([commit](https://github.com/istio/istio/pull/43012/commits/c67590471f97df118ab5819b56e1a1c9ae70da14)) and that was throwing off my tests